### ISSUE TITLE
Battleborn dpks

### ISSUE THREAD

**Issue by lowlines** [gildor2/UModel#8]
Follow up of this discussion: http://www.gildor.org/smf/index.php?topic=2987.0
dpks are basically tfcs with a different header structure and are tied to a upk with the same filename.

```
header {
int headerSize,
int tableSize,
int tableCount,
<tableCount> headerTable,
<tableCount> entryData
}
headerTable {
int fileOffset, // from start of file
int checksum?,
int compType?, // have only seen value of 1 so far, compSize and uncompSize will match
int compSize?,
int uncompSize?
}
entryData {
<0x35> unknownData
}
```

This will correctly point to the start of each texture file cache entry which begins with DD7478BB. Hopefully this helps with adding support for them.

**Comment by gildor2** [comment]
According to discussion on the forum, this game has TFC files. So, what DPKs are for? Do you have any information?

**Comment by lowlines** [comment]
My impression is that instead of having a single TFC file, they've decided to split it into smaller DPK files for a lot of UPKs. The normal Textures.tfc is only 159MB while there are DPKs which are larger than that. The total size of all DPKs ends up at about 7.46GB so perhaps it is for performance?

**Comment by gildor2** [comment]
Many companies does that. Even UE4.12 now has possibility to organize texture data in similar (but more flexible) way.

**Comment by lowlines** [comment]
I have been able to extract all non compressed textures from dpks however, how do Texture2Ds determine the offset into a texture cache file? ie I have 2 Texture2Ds which are identical save for the first 4 bytes (which I assume is an ID) which are 44 and 50 respectively, however the actual texture cache blocks are found at index 650 and 651 in the dpk. How do normal Texture2D/TFCs do this? I've tried stepping through umodel's source code but I can't see where the offset is being determined.
